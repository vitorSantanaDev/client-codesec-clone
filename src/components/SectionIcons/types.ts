import { ISectionIconsProps } from 'interfaces/sections'

export type IViewProps = ISectionIconsProps
