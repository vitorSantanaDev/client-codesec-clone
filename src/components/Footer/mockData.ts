export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
