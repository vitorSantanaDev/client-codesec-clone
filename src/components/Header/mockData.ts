import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
