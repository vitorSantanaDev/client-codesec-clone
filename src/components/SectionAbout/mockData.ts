export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
