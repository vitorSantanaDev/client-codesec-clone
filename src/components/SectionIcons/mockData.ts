import { IViewProps } from './types'

export const mockIconsProps: IViewProps = {
  _id: String(Math.random() * 1000),
  icons: [
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    },
    {
      _id: String(Math.random() * 10000),
      image: {
        urlImage: '/images/icon-calendar.svg',
        altText: 'Icone de calendario'
      },
      textInfo: 'Exclusividade no mercado'
    }
  ]
}
