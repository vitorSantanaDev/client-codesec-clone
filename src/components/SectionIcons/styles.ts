import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
  `}
`
export const Icon = styled.div`
  ${({ theme }) => css`
    text-align: center;
    margin-bottom: ${theme.spacings.small};
  `}
`
export const Text = styled.p`
  ${({ theme }) => css`
    width: 100%;
    max-width: 120px;

    color: ${theme.colors.whiteDark};
    font-size: ${theme.font.sizes.small};
  `}
`
