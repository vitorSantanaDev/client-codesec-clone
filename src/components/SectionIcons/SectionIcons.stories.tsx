import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionIcons from '.'
import { mockIconsProps } from './mockData'

export default {
  title: 'SectionIcons',
  component: SectionIcons,
  args: {
    ...mockIconsProps
  }
} as ComponentMeta<typeof SectionIcons>

export const Default: ComponentStory<typeof SectionIcons> = (args) => (
  <SectionIcons {...args} />
)
