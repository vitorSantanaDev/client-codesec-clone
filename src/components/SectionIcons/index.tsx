import Container from 'components/Container'

import { IViewProps } from './types'
import * as S from './styles'

const SectionIcons = ({ icons }: IViewProps) => (
  <Container>
    <S.Wrapper>
      {icons.map((icon) => {
        return (
          <S.Icon data-aos="fade-up" key={icon._id}>
            <img src={icon.image.urlImage} alt={icon.image.altText} />
            <S.Text>{icon.textInfo}</S.Text>
          </S.Icon>
        )
      })}
    </S.Wrapper>
  </Container>
)

export default SectionIcons
