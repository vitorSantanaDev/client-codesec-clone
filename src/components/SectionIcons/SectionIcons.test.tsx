import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionIcons from '.'
import { mockIconsProps } from './mockData'

describe('<SectionIcons />', () => {
  it('should render icons with description', () => {
    const { container } = renderTheme(<SectionIcons {...mockIconsProps} />)
    const icons = screen.getAllByRole('img', {
      name: mockIconsProps.icons[0].image.altText
    })
    const description = screen.getAllByText(/Exclusividade no mercado/i)
    expect(icons).toHaveLength(7)
    expect(description).toHaveLength(7)
    expect(container.firstChild).toMatchSnapshot()
  })
})
