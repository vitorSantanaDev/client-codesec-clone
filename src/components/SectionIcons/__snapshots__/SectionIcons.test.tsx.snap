// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionIcons /> should render icons with description 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 {
  text-align: center;
  margin-bottom: 1.6rem;
}

.c3 {
  width: 100%;
  max-width: 120px;
  color: #EDE4E5;
  font-size: 1.6rem;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <img
        alt="Exclusividade no mercado"
        src="/images/icon-calendar.svg"
      />
      <p
        class="c3"
      >
        Exclusividade no mercado
      </p>
    </div>
  </section>
</div>
`;
